### ISSUE TITLE
Feature request: Support for Windows 10

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/sudo#2]
We should backport Sudo to Windows 10. Ultimately, this shouldn't be too hard, but porting the settings app changes may be a bit of a pain.

This is internally, MSFT:48735673

**Comment by Jevil7452** [comment]
Sudo just runs on Windows 10. Configuration can be done via the command line tool itself.
![image](https://github.com/microsoft/sudo/assets/98893064/e17a6aaf-a3f1-49f0-87c9-888e9ea3d8ae)
So, that should clear up the first part of the problem. But I guess you would like to have a friendly config UI for it.

**Comment by zadjii-msft** [comment]
Yep! `sudo.exe` doesn't do anything Windows 11-specific. The Settings app changes are the hardest part (since there was a pretty massive UI rejuv between 10 and 11).

**Comment by 3052** [comment]
OK, where do you even get it? its not here:
https://aka.ms/sudo

or here:
https://github.com/microsoft/sudo

or here:
https://github.com/microsoft/sudo/releases

are we really expected to install Windows 11 to test this on Windows 10?

**Comment by zadjii-msft** [comment]
When we get this backported to the Windows 10 branches, you can make sure we'll post here 😄

**Comment by gerardog** [comment]
I see sudo.exe is (currently) portable and works on Windows 10. So why ship it with windows? Will not ship it as a Winget package?

**Comment by gerardog** [comment]
I see sudo.exe is (currently) portable and works on Windows 10. So why ship it with windows? Why not ship it as a Winget package?

**Comment by 3052** [comment]
$

**Comment by zadjii-msft** [comment]
> Why not ship it as a Winget package?

That's certainly something to look at going forward. I'd want to make sure that the GPO stuff we have in-box with the OS is also something that could be delivered via `winget`. Otherwise, we'd have to consider the scenario where someone can install sudo, but then there's no way to manage it from an organization's standpoint. Something for us to work out in the coming months ☺️

**Comment by LeonardoIz** [comment]
You can download Sudo from [here](https://github.com/LeonardoIz/sudo-builds), it doesn't have an installer yet, but I'm working on it.

**Comment by rbarbrow** [comment]
Any update on the winget front?

**Comment by zadjii-msft** [comment]
Okay I definitely get that it's valuable - but I'm gonna go ahead and delete links to 3p builds of sudo. Sorry, I just think it's a terrible idea to have folks downloading builds from potentially untrusted third parties. Seems like a recipe for a supply chain vulnerability. 

I'm back from vacation this week[^1]. Getting officially signed installers up on our releases page and `winget` is my priority this month. 

[^1]: though, it's a pair of short weeks for me. So, things will be a bit slower than usual

**Comment by LeonardoIz** [comment]
The original code is not modified, it is compiled by cloning the original
repository, you can check the github action

**Comment by zadjii-msft** [comment]
IMO: Doesn't matter if it's not changed _now_. In the wake of the `xz` thing, I'm not gonna have folks take any chances on untrusted third-parties. Even ones that are well intended now may well later become compromised.

**Comment by Neustradamus** [comment]
It will be nice to have for Windows 10 and Windows Server too.

**Comment by git-78** [comment]
We are now 11 months after. Any news? What's about to publish an exe version, here or through winget ?

**Comment by sebma** [comment]
> Sudo just runs on Windows 10. Configuration can be done via the command line tool itself. ![image](https://github.com/microsoft/sudo/assets/98893064/e17a6aaf-a3f1-49f0-87c9-888e9ea3d8ae) So, that should clear up the first part of the problem. But I guess you would like to have a friendly config UI for it.

@Jevil7452 I see [no sudo github release there](https://github.com/microsoft/sudo/releases) : how did you "install" this `sudo` on Windows 10 ?

**Comment by Jevil7452** [comment]
Take the EXE from Windows 11.

**Comment by sebma** [comment]
@Jevil7452 OK Thanks, I'll do this tomorrow : 
```powershell
copy $(gcm sudo).source -PassThru | Compress-Archive -DestinationPath sudo-W11.zip -f
rm sudo.exe
```

**Comment by sebma** [comment]
@Jevil7452 As you suggest, I just copied `sudo.exe` from Windows 11 into my Windows 10 and here is what I get : 

<img width="1186" height="159" alt="Image" src="https://github.com/user-attachments/assets/57884b0a-c18a-47f1-ae96-4bf6e10a5614" />

```text
Sudo is disabled on this machine. To enable it, go to the Developer Settings page in the Settings app
```
But I see no `Enable sudo` parameter in the Developer Settings of my Windows 10 : 

<img width="1600" height="857" alt="Image" src="https://github.com/user-attachments/assets/fae09043-2b90-4ad6-9b5f-eafc62ded561" />

**Comment by plutonium-239** [comment]
Maybe this can help? : https://www.elevenforum.com/t/enable-or-disable-sudo-command-in-windows-11.22329/
It includes 2 methods other than the settings UI (through the CLI and through the registry)

**Comment by sebma** [comment]
We have two users on Windows : a standard user and an admin user.

After typing these 2 commands as an administrator : 
```powershell
PS C:\WINDOWS\system32> sudo config --enable enable
Sudo is currently in Inline mode on this machine
PS C:\WINDOWS\system32> sudo config --enable forceNewWindow
Sudo is currently in Force New Window mode on this machine
PS C:\WINDOWS\system32>
```
And now I get this when I get back to the normal user (another user) : 
```powershell
PS C:\> sudo cmd
You are not allowed to run sudo
PS C:\>
```
Unfortunenatly, there is no ` -u | --user {username} Run as the specified user. Asks for password. For local admins shows UAC unless '-i Medium'` option like in the [`gerardog gsudo`](https://github.com/gerardog/gsudo).

Related #127

**Comment by plutonium-239** [comment]
I guess that depends on the implementation itself, saw the issue. 

If you used it as the user with admin privileges but from a non-admin shell, does it let you escalate?

**Comment by sebma** [comment]
> If you used it as the user with admin privileges but from a non-admin shell, does it let you escalate?

@plutonium-239 Didn't have the time, they had already schedule the upgrade of my laptop upto Windows 11.

Maybe someone else with a Windows 10 can test what you asked.

**Comment by Jy-EggRoll** [comment]
Any updates? The experience of publishing and retrieving sudo using winget should be quite good, which could simplify the operation of many Windows CLI programs.

**Comment by iamqiz** [comment]
> Yep! `sudo.exe` doesn't do anything Windows 11-specific. The Settings app changes are the hardest part (since there was a pretty massive UI rejuv between 10 and 11).

@zadjii-msft  why not release the first version sudo for win10 without any UI settings? it is a CLI tool, I think no need to design settings GUI for it,it should be configured via the command line.
